Initial draft for the security policy (#4089)
timescale/timescaledb: SECURITY.md

# Security Policy

We aim to keep Timescale safe for everyone. 
Publicly disclosing bugs in a public forum puts everyone in the Timescale community at risk.

## Supported Versions

> Please take a look at our [Support Policy](https://www.timescale.com/legal/support-policy).

Author: ```suggestion
The supported version is always the latest major release available in our repository.
We also release regular minor versions with fixes and corrections alongside some new features as well as patchfix releases, that you should keep upgrading to.
 
You can also take a look at our [Support Policy](https://www.timescale.com/legal/support-policy).
```

# Security Policy

> We aim to keep Timescale safe for everyone. 

Reviewer: ```suggestion
We aim to keep TimescaleDB safe for everyone. 
```

Reviewer: Since this is in the timescaledb repo, I would be specific.

> Publicly disclosing bugs in a public forum puts everyone in the Timescale community at risk.

Reviewer: "bugs" can be other things than vulnerabilities. 
Maybe we keep it simple and remove this entirely. We already have the section on how to report a vulnerability.

Author: fixed.

## Supported Versions

The supported version is always the latest major release available in our repository.

> We also release regular minor versions with fixes and corrections alongside some new features as well as patchfix releases, that you should keep upgrading to.

Reviewer: ```suggestion
We also release regular minor versions with fixes and corrections alongside some new features as well as patchfix releases, that you should keep upgrading to.
Vulnerability fixes are made available as part of these patchfix releases.
```

Reviewer: Instead of "patchfix" release, consider using "patch" release following the semver.org format.

# Security Policy

We aim to keep TimescaleDB safe for everyone. 

> Publicly disclosing bugs in a public forum puts everyone in the Timescale community at risk.

Author: ```suggestion
Publicly disclosing security bugs in a public forum puts everyone in the Timescale community at risk.
```

> Publicly disclosing security bugs in a public forum puts everyone in the Timescale community at risk.

Reviewer 2: ```suggestion
Publicly disclosing security bugs in a public forum can put everyone in the Timescale community at risk,
however. Therefore, we ask that people follow the below instructions to report security vulnerability.
The entire Timescale community thanks you!
```

> ## Supported Versions

Author: @jfjoly do you have any other ideas?

> ## Supported Versions

Reviewer 2: I again would cut this.  This "support policy" referenced is about paid commercial support, and the "supported version" you reference here refers to commercial support guarantees.  This is about contractual obligations, and have nothing to do with our open-source/community promises.  My feeling is that it will just raise fear/uncertaintly/doubt, for little benefit.d

> ## Supported Versions

Reviewer: I checked the default policy template that Github provides (screenshot below). 
The goal of this section is to indicate which versions receive security fixes.

We could reword it with examples and be explicit that supported means providing security fixes: 
> We provide security fixes for the latest version of TimescaleDB. Minor versions (2.1 , 2.2...) don't get security fixes once a newer minor version is released. 
> For example, 2.6.0 is the most recent versions as of March 2022. New vulnerabilities will only be fixed against 2.6, there will be no backport to 2.5 or any older version.  

  
<img width="890" alt="Screen Shot 2022-03-23 at 18 32 26" src="https://user-images.githubusercontent.com/7732950/160561623-2dbaf1ec-1dc5-4263-9439-dc95c7121972.png">

> ## Supported Versions

Author: There was a previous discussion and this is why it was removed.

# Security Policy

We aim to keep Timescale safe for everyone. 
Publicly disclosing bugs in a public forum puts everyone in the Timescale community at risk.

> ## Supported Versions

Reviewer 3: Is there another place we call out supported versions? It would be best to link out to that than have this become unmaintained.

Author: See my comment bellow.

> 
> Our users should always upgrade to the latest version.
> 
> | Version | Supported          |
> | ------- | ------------------ |
> | 2.6.x   | :white_check_mark: |
> | 1.x   | :x:                |

Author: ```suggestion

Please take a look at our [Support Policy](https://www.timescale.com/legal/support-policy).
```

@clarkbw here.

Reviewer 3: > https://www.timescale.com/legal/support-policy

I don't think this page helps anyone with the question of "is this release where I found an issue a supported version?". I didn't want us to have this page refer to `2.6.x` when we're on to `3.1.0` because it could create confusion.  I was hoping for some page in the repo with the latest supported version list.  Or perhaps we could say the latest major version published in the repo is supported. Not sure.

Author: This was the original suggestion.

Reviewer 3: The original suggestion was to include the supported versions in this page but that creates a place where we need to ensure we have updated the supported version as new versions are released. If we don't have a place where we declare supported versions I'm not sure what we should do next, but having a page like that might be an approach.

## Reporting a Vulnerability

> If you find a vulnerability in our software the e-mail address to use to contact the Timescale Security Team is security@timescale.com.

Reviewer 4: Some thoughts about the email message:
- maybe we can add some hints what information we would like to see in the email (type of the vulnerabilty, possible impact, repro steps)
- we can provide a PGP key to encrypt the mail
- response SLA

# Security Policy

We aim to keep TimescaleDB safe for everyone. 
Publicly disclosing security bugs in a public forum can put everyone in the Timescale community at risk,
however. Therefore, we ask that people follow the below instructions to report security vulnerability.
The entire Timescale community thanks you!

## Supported Versions

The supported version is always the latest major release available in our repository.
We also release regular minor versions with fixes and corrections alongside some new features as well as patchfix releases, that you should keep upgrading to.

> Vulnerability fixes are made available as part of these patchfix releases.

Author: ```suggestion
Vulnerability fixes are made available as part of these patchfix releases and you can read our list of [Security Advisories](https://github.com/timescale/timescaledb/security/advisories?state=published).
```
